system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -79,11 +83,51 @@ this.TopStoriesFeed = class TopStoriesFeed {
       this.storiesLastUpdated = Date.now();
       // This is filtered so an update function can return true to retry on the next run
       this.contentUpdateQueue = this.contentUpdateQueue.filter(update => update());
+
+      this.setCache(STORIES_CACHE_KEY, this.stories);
     } catch (error) {
       Cu.reportError(`Failed to fetch content: ${error.message}`);
     }
   }
 
+  loadCachedStories() {
+    this.stories = this.getCachedStories();
+    if (this.stories && this.stories.length > 0) {
+      this.dispatchUpdateEvent(this.storiesLastUpdated, {rows: this.stories});
```

Comment:
This would cause 2 dispatches early on: one from cache and one from network. I think that should be okay… @csadilek?

Reply:
Yes, that would be OK as the second one isn't broadcasting, but we shouldn't assign to ```this.stories``` as this will affect the spoc experiment and we will get rid of ```this.stories``` soon. Let's just assign ```const stories = this.getCachedStories()```.